pom.xml

```diff
@@ -545,6 +545,7 @@
        =================================================================== -->
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <project.build.outputTimestamp>2023-07-15T16:42:39Z</project.build.outputTimestamp>
```

Comment:
Thanks. But what would be the maintenance mechanism? Would it be necessary to update this line manually during merges?

Alternatively (I know it is a much larger debate), I wonder why reproductive build proponents insist on bit-by-bit reproducibility instead of creating a `jar-diff` tools which ignores the order of entries in the JAR files, and ignores a small selected set of `META-INF/MANIFEST.MF` entries such as the build timestamp?

Reply:
on the maintenance mechanism: it can be maven-release-plugin, or versions-maven-plugin. Or if you prefer, you can use the git commit date.
Honestly, merging is not really something that happens, given the timestamp is only updated usually at most during version changes (between SNAPSHOT and release)

on why binary check vs smart tools:
- smart tools will have bugs
- efficiency (it's not only about timestamp, but also order)
- particularly in the case of cache mechanisms

Reply:
A proposal to migrate from Maven to Gradle is [on the table](https://lists.apache.org/thread/xqnydv188cxom22forz3j8bbkbtv8g14) in order to modularize the project with `module-info.java` files. While Maven has some JPMS support, taking full advantage of JPMS requires restructuring the code from _Package Hierarchy_ to _Module Source Hierarchy_, and the latter would be very difficult to do with Maven. If this migration is accepted, the fix proposed in this issue would take a different form.

On binary check versus smart tools, the problem of binary check is that it throws away valuable metadata such as build time and who made the build. While the metadata seems to be there, they became meaningless in their values are not anymore what they seem to be. Metadata with false values are worst than no metadata, because they create a situation where we don't know anymore which metadata we can trust. In the particular case of this issue, I think that no timestamp would be better than a timestamp with fixed value. I highly value the goal of reproductible builds (it would make me feel safer when I deploy binaries on Maven Central), but I think that an extremist position on "bit-by-bit" reproducibility is creating other dangers in a world where false information became a thread as serious as hackers. In French I would said that bit-by-bit reproducibility is "jeter le bébé avec l'eau du bain".

Reply:
> If this migration is accepted, the fix proposed in this issue would take a different form.

ok, let's drop this PR, then, no problem

> it throws away valuable metadata such as build time and who made the build

I perfectly understand your idea, I was a strong promoter of trackers of provenance: the fact is that with RB, they are not useful any more

> I think that no timestamp would be better than a timestamp with fixed value

Yes: if you know if zip and/or tar have that notion, I'm interested

> In French I would said that bit-by-bit reproducibility is "jeter le bébé avec l'eau du bain".

Hehe :) From experience gained from thousands of rebuilds in https://github.com/jvm-repo-rebuild/reproducible-central , I detected many unexpected difference in releases with this approach
But it has to happen step by step, learning while deploying, without forcing